### ISSUE TITLE
"Units defined" check

### ISSUE THREAD

**Issue by gothub** [NCEAS/metadig-engine#60]
The check that will have the id "check.units.defined.1" (currently line 38 of [MetaDIG Suite Planning](https://docs.google.com/spreadsheets/d/1771E8b5PbnyBKKtYYaz-D4GeTqO2WwONI5LndaECe1s/edit#gid=0)) is slated to "Check that masurement scale is present and correct".

Is this correct? If we are checking `measurementScale`, EML defines 5 types, but what do we
check for them:
- nominal
   - this is a user defined scale of classes
- ordinal
   - user defined catagories, e.g. "low", "medium", "high"
- interval
  - ordinal scale with equally sized unit, e.g. "Celcisus"
- ratio
- dateTime
   - ISO 8601 formatted dates with legacy exceptions (defined by EML standard)

This is listed as an "Interpretation and Use" check - should this be listed as "Data Congruency", and we check if the data matches the scales defined?

For all the of complex checks like this, do we just SKIP if it is not clear what to for a given
dataset attribute?

**Comment by amoeba** [comment]
> what do we check for them:

This is a super hard check to write. Maybe one of the hardest ones. 

At the very least, it should be checking that every attribute has a measurement scale with some reasonable content. For example, when a nominal scale is used, if that attribute had one or more `codeDefinition`s, each with non-zero-length and non-identical `code`s and `definition`s, I would consider that attribute as having a scale. Of course this covers like 1/100th of the ways units can get put onto an attribute and I haven't even written you any code or xpaths so I'm not helping that much.

I'm happy to help work on this one.

> This is listed as an "Interpretation and Use" check - should this be listed as "Data Congruency", and we check if the data matches the scales defined?

I think that this should be coded up to just check the units in the EML and, if we want to also check the congruency between the units in the EML and the data, we can write another check. We should be able to re-use a lot of code between the checks and/or even wrap up some common functionality in a helper lib.

**Comment by gothub** [comment]
@amoeba Yep, your suggestion for `nominal` is just what we need - a fairly simple check
that will represent a reasonable percentage of the datasets checked. 

We could have separate checks for each `measurementScale` type, and just SKIP them if
that type isn't present.

**Comment by amoeba** [comment]
That's not a bad idea. I think I prefer one Check being 'All attributes have units of some reasonable form' as a general check that gets at the use/interp use case. No matter what we do, it seems like we can always tweak the Check(s) once someone has done the research on XPaths and related code.
